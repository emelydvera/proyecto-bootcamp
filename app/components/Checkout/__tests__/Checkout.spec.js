const React = require('react');
const { screen, render } = require('@testing-library/react');
const { within } = require('@testing-library/dom');
const Checkout = require('..');
const { data } = require('../../../../mocks/test/get/https/api.mercadolibre.com/items/MLA1104983845.json');

describe('Checkout component', () => {

    it('should render correctly', () => {
        const props = {
            product: data,
            quantity: '1',
        };
        const { asFragment } = render(<Checkout {...props} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('total price should be unit price multiplied by stock quantity available when products quantity selected is higher than the stock quantity available', () => {
        const props = {
            product: data,
            quantity: '2',
        };
        render(<Checkout {...props} />);
        const moneyAmount = document.querySelector('#checkout > div:nth-child(6) > span');
        let result = within(moneyAmount).getByText('30.999');
        expect(result).toHaveTextContent('30.999');
    });

    it('must display a shortened version of the title if it is longer than 70 characters', () => {
        const props = {
            product: {
                ...data,
                title: 'Apple Macbook Air (13 Pulgadas, 2020, Chip M1, 256 Gb De Ssd, 8 Gb De Ram) - Gris Espacial',
            },
            quantity: '1',
        };

        render(<Checkout {...props} />);
        const title = screen.getByRole('heading', {
            name: /apple macbook air \(13 pulgadas, 2020, chip m1, 256 gb de ssd, 8 gb \.\.\./i
        });
        expect(title).toHaveTextContent(/apple macbook air \(13 pulgadas, 2020, chip m1, 256 gb de ssd, 8 gb \.\.\./i);
    });

    it('must show shipping with cost to calculate if the product does not have free shipping', () => {
        const props = {
            product: {
                ...data,
                shipping: { free_shipping: false },
            },
            quantity: '1',
        };
        render(<Checkout {...props} />);
        const shipping = screen.getByText(/envío con costo a determinar/i);
        expect(shipping).toHaveTextContent(/envío con costo a determinar/i);

    });
});