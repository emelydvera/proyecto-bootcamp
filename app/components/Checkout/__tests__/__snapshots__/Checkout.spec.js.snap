// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="checkout"
    id="checkout"
  >
    <h2
      aria-label="Resumen de compra"
      class="checkout__title"
      tabindex="9"
    >
      Resumen de compra
    </h2>
    <p
      class="checkout__subtitle"
    >
      Producto
    </p>
    <div
      class="checkout__product"
    >
      <img
        alt=" Moto G20 64 Gb  Azul Cielo 4 Gb Ram"
        class="image lazy-loadable"
        data-src="http://http2.mlstatic.com/D_852458-MLA48270995220_112021-O.jpg"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        tabindex="10"
      />
      <div
        class="column"
      >
        <div
          class="checkout__product__container"
        >
          <h2
            tabindex="11"
          >
             Moto G20 64 Gb  Azul Cielo 4 Gb Ram
          </h2>
          <span
            aria-label="precio producto"
            class="checkout__product__unityprice"
            tabindex="13"
          >
            1x
            <span
              class="andes-money-amount andes-money-amount--cents-superscript"
              style="font-size: 16px;"
            >
              <span
                class="andes-visually-hidden"
              >
                30999 pesos
              </span>
              <span
                aria-hidden="true"
                class="andes-money-amount__currency-symbol"
              >
                $
              </span>
              <span
                aria-hidden="true"
                class="andes-money-amount__fraction"
              >
                30.999
              </span>
              <span
                aria-hidden="true"
                class="andes-visually-hidden"
              >
                ,
              </span>
              <span
                aria-hidden="true"
                class="andes-money-amount__cents andes-money-amount__cents--superscript-16"
                style="font-size: 10px; margin-top: 2px;"
              >
                00
              </span>
            </span>
          </span>
        </div>
        <div
          class="checkout__input__container"
        >
          <div
            class="input__quantity"
          >
            <div
              class="andes-form-control andes-form-control--textfield andes-form-control--centered andes-form-control--default checkout__input"
            >
              <label>
                <span
                  class="andes-form-control__label"
                >
                  Cantidad
                </span>
                <div
                  class="andes-form-control__control"
                >
                  <input
                    aria-label="Ingrese la cantidad de productos a comprar"
                    class="andes-form-control__field"
                    maxlength="120"
                    rows="1"
                    tabindex="12"
                    type="number"
                    value="1"
                  />
                </div>
              </label>
              <div
                class="andes-form-control__bottom"
              >
                <span
                  class="andes-form-control__message"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <p
      class="checkout__subtitle"
    >
      Envío
    </p>
    <div
      aria-label="Costo de envío"
      class="row checkout__product__info envio"
    >
      <div
        class="envio__gratis"
      >
        <svg
          fill="green"
          height="24"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            clip-rule="evenodd"
            d="M5.25 5.25H17.3539C17.7681 5.25 18.1039 5.58579 18.1039 6C18.1039 6.08759 18.0885 6.1745 18.0586 6.25679L15.1227 14.3133C14.0072 14.548 13.1198 15.4023 12.8375 16.4995H8.1585C7.88715 15.445 7.05698 14.6149 6.0025 14.3435V14.249H0.75V15.749H2.65356C2.39817 16.1904 2.252 16.7029 2.252 17.2495C2.252 18.9066 3.59537 20.25 5.2525 20.25C6.65066 20.25 7.82546 19.2937 8.1585 17.9995H12.8375C13.1705 19.2937 14.3453 20.25 15.7435 20.25C17.1417 20.25 18.3165 19.2937 18.6495 17.9995H21.5077L23.2984 13.5235L21.5406 8.25H18.9287L19.4679 6.77037C19.5579 6.52349 19.6039 6.26276 19.6039 6C19.6039 4.75736 18.5965 3.75 17.3539 3.75H5.25V5.25ZM6.753 17.2495C6.753 16.4208 6.0812 15.749 5.2525 15.749C4.4238 15.749 3.752 16.4208 3.752 17.2495C3.752 18.0782 4.4238 18.75 5.2525 18.75C6.0812 18.75 6.753 18.0782 6.753 17.2495ZM18.6495 16.4995C18.395 15.5105 17.649 14.7189 16.6874 14.4005L18.3821 9.75H20.4594L21.7016 13.4765L20.4922 16.4995H18.6495ZM14.243 17.2495C14.243 18.0782 14.9148 18.75 15.7435 18.75C16.5722 18.75 17.244 18.0782 17.244 17.2495C17.244 16.4208 16.5722 15.749 15.7435 15.749C14.9148 15.749 14.243 16.4208 14.243 17.2495Z"
            fill="green"
            fill-rule="evenodd"
          />
          <path
            d="M11.252 6.75H3.753V8.25H11.252V6.75Z"
            fill="green"
          />
          <path
            d="M0.75 9.75V11.25H6.75V9.75H0.75Z"
            fill="green"
          />
        </svg>
        <p
          class="envio__gratis--true"
          tabindex="14"
        >
          Envío gratis
        </p>
        <p
          class="envio__gratis__city "
          tabindex="14"
        >
          desde Balvanera
        </p>
      </div>
    </div>
    <div
      aria-label="Precio total"
      class="checkout__total"
      tabindex="15"
    >
      <p
        class="checkout__total__price"
      >
        Total
      </p>
      <span
        class="andes-money-amount andes-money-amount--cents-superscript"
        style="font-size: 28px;"
      >
        <span
          class="andes-visually-hidden"
        />
        <span
          aria-hidden="true"
          class="andes-money-amount__currency-symbol"
        >
          $
        </span>
        <span
          aria-hidden="true"
          class="andes-money-amount__fraction"
        >
          30.999
        </span>
        <span
          aria-hidden="true"
          class="andes-visually-hidden"
        >
          ,
        </span>
        <span
          aria-hidden="true"
          class="andes-money-amount__cents andes-money-amount__cents--superscript-28"
          style="font-size: 14px; margin-top: 6px;"
        >
          00
        </span>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
